app/api/auth.py

```diff
@@ -0,0 +1,134 @@
+    hashed_password: str
+
+
+pwd_context = CryptContext(schemes=['bcrypt'], deprecated='auto')
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl='token')
+
+router = APIRouter()
+
+session = Session(bind=engine)
+
+
+def create_access_token(data: dict, expires_delta: Optional[timedelta] = None):
+    to_encode = data
+    if expires_delta:
+        expire = datetime.utcnow() + expires_delta
+    else:
+        expire = datetime.utcnow() + timedelta(minutes=15)
+    to_encode.update({'exp': expire})
+    encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
+    return encoded_jwt
+
+
+def get_user_exception():
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    return credentials_exception
+
+
+def get_active_user_exception():
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_400_BAD_REQUEST,
+        detail="Inactive user",
+    )
+    return credentials_exception
+
+
+def token_exception():
+    token_exception_response = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Incorrect username or password",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    return token_exception_response
+
+
+async def get_current_user(token: str = Depends(oauth2_scheme)):
+    credentials_exception = get_user_exception()
+    try:
+        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
+        username: str = payload.get('sub')
+        if username is None:
+            raise credentials_exception
+        token_data = TokenData(username=username)
+    except JWTError:
+        raise credentials_exception
+    user = _get_user(username=token_data.username)
+    if user is None:
+        raise credentials_exception
+    return user
+
+
+async def get_current_active_user(current_user: User = Depends(get_current_user)):
+    if current_user.disabled:
+        raise get_active_user_exception()
+    return current_user
+
+
+@router.post('/token', response_model=Token)
+async def login(form_data: OAuth2PasswordRequestForm = Depends()):
+    user = _authenticate_user(form_data.username, form_data.password)
+    if not user:
+        raise token_exception()
+    access_token_expires = timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
+    access_token = create_access_token(
+        data={'id': user.id}, expires_delta=access_token_expires
+    )
+    return {'access_token': access_token, 'token_type': 'bearer'}
+
+
+def _verify_password(plain_password, hashed_password):
+    return pwd_context.verify(plain_password, hashed_password)
+
+
+def _get_password_hash(password):
+    return pwd_context.hash(password)
+
+
+def _get_user(username: str):
+    db = session.query(User).filter(User.username == username).first()
+    if username in db:
+        user_dict = db[username]
+        return UserInDB(**user_dict)
```

Comment:
```suggestion
    user = session.query(User).filter(User.username == username).first()
    if username in user:
        user_dict = user[username]
        return UserInDB(**user_dict)
```
p1: 호출된 대상은 db 가 아니라 user 라서 변경해 주시는 것이 좋을것 같습니다. 
ask: username 은 db 에서 불러 오는 것이 아닌가요?